### ISSUE TITLE
Add accessible interface to event icons

### ISSUE THREAD

**Issue by pmartz** [thewca/worldcubeassociation.org#10454]
I’m a blind cuber and solve using a tactile cube. When I access the WCA website, I depend on screen reader software. On a desktop or laptop, I access all links and buttons using a keyboard, not a mouse. Website elements are verbally announced via a speaker or earbuds. As a result, it’s important that the WCA website be designed and implemented with accessibility in mind.

The event icons are not accessible using the VoiceOver screen reader on MacOS regardless of browser (I’ve tested Firefox, Safari, and Chrome). They don’t show up as links or as form controls, making them nearly impossible to locate. If I am able to locate them (say, with the help of someone sighted), navigating through the controls produces no audible feedback, making them impossible to differentiate. They do not respond to the VoiceOver selection keyboard shortcut, and the only way to activate them is to bring the mouse pointer to the icon and click the mouse. In short, they can only be used by fully sighted users.

The situation isn’t much better in iOS with VoiceOver and Safari. Again, the name of each icon is not announced, making them impossible to differentiate. But, if someone is looking over my shoulder and telling me which icon is which, at least I can select them.

These icons are used all over the website, from event selection during registration to viewing competition results to viewing my own individual scores. As currently implemented, these icons prevent me from performing any of these tasks independently.

My request is that you enhance these existing event icons to include standard HTML links or buttons that are not solely image-based. I’m requesting that this change be made globally throughout the WCA website in every place that these icons are used.

 In my head, I imagine the existing event controls would still be present with accessible links or buttons alongside them, and users could select using whichever interface they prefer. With such an implementation, existing UI code wouldn’t need to be altered, only supplemented with an alternate accessible interface.

Implementing this feature would make the WCA website usable and accessible. It would bring it into compliance with the Americans with Disabilities Act. It would make me feel included in spite of my disability. My sighted spouse is getting pretty tired of coming over to my computer or phone and clicking these icons on my behalf. Please help! Thanks.
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by jfly** [comment]
@pmartz, I'm a maintainer of cubing icons, which the WCA website uses under the hood. How awful is our homepage for you? https://icons.cubing.net/. Specifically the area after "It's time for cubing icons!" and right before "Cubing icons can be used for any cubing project".

Do you know if this would get better for you if we added an `alt` tag?

**Comment by pmartz** [comment]
Thanks for looking at this.

I opened your website. Using iOS with Safari and VoiceOver, each cube icon is announced as “in-page link”. Similar in MacOS with Chrome and VoiceOver, where each icon is announced as “link”. In both cases, it is essentially impossible to tell one icon from another. I believe adding ALT text to the icons is worth a try. If you make a change and want me to test, let me know. I totally appreciate your help on this.

Paul Martz
PaulMartz.com <https://paulmartz.com/>
